src/main/java/org/openlmis/stockmanagement/controller/StockCardController.java

```diff
@@ -398,5 +452,24 @@ private void removeEmptyLotsFromStockCard(StockCard stockCard)
         stockCard.setLotsOnHand(nonEmptyLots);
     }
 
+    private List<Right> getRightsForUserFacilityAndProductCode(Long userId, Long facilityId, String productCode)
```

Comment:
It may be worth putting this code block (or even the entire getRightsForUserFacilityAndProductCode function) in a service in order to make it more reusable. ProgramService seems like perhaps the ideal spot. However, the code uses ProgramProductService as well as ProgramService, and the former already has a dependency on later. Putting this code in ProgramProductService would thus alleviate adding a circular-dependency to ProgramService.